include/cppkafka/utils/buffered_producer.h

```diff
@@ -519,17 +518,18 @@ class CPPKAFKA_API BufferedProducer {
         return nullptr;
     }
     template <typename BuilderType>
-    void do_add_message(BuilderType&& builder, MessagePriority priority, bool do_flush);
+    void do_add_message(BuilderType&& builder, bool is_retry, bool do_flush);
```

Comment:
[suggestion for readability] I think that using too many bools in the function signature makes it hard to understand when you read code. What is `do_add_message(builder, true, false)`? or `do_add_message(builder, false, false)`? difficult to tell unless you know the parameters beforehand. I would be ideal to have enums and use something like:
`do_add_message(builder, QueueType::RetryQueue, FlushAction::flush)`

This would also apply to `void async_produce(BuilderType&& message, bool throw_on_error)` where the `throw_on_error` would be made into an enum like `ThrowAction::OnError` or something like that. I would leave this to @demin80 to decide.